### PR TITLE
Close unclosed files

### PR DESCRIPTION
**Pull request by CallanTaylor**

### CODE REVIEW

File: eBook/examples/chapter_12/read_files.go [hunk 1]
```diff
@@ -20,6 +20,7 @@ func main() {
 			fmt.Printf("The file %v does not exist!\n", flag.Arg(i))
 			break
 		}
+		defer fin.Close()
```
Comment (unknwon) [inline eBook/examples/chapter_12/read_files.go]: `defer` in a `for` loop is not ideal, should just have it closed as the last statement in the loop (i.e. after line 27).

Comment (CallanTaylor) [reply eBook/examples/chapter_12/read_files.go]: Great point. Updated now

File: eBook/examples/chapter_12/cat.go [hunk 2]
```diff
@@ -30,6 +30,7 @@ func main() {
 			fmt.Fprintf(os.Stderr, "%s:error reading from %s: %s\n", os.Args[0], flag.Arg(i), err.Error())
 			continue
 		}
+		defer f.Close()
```
Comment (unknwon) [inline eBook/examples/chapter_12/cat.go]: Sorry I missed this one, it's another "defer in a loop".

File: eBook/exercises/chapter_12/cat_numbered.go [hunk 3]
```diff
@@ -38,6 +38,7 @@ func main() {
 			fmt.Fprintf(os.Stderr, "%s:error reading from %s: %s\n", os.Args[0], flag.Arg(i), err.Error())
 			continue
 		}
+		defer f.Close()
```
Comment (unknwon) [inline eBook/exercises/chapter_12/cat_numbered.go]: Sorry I missed this one, it's another "defer in a loop".

**Pull request merged**